### PR TITLE
Refactor Dataproc Trigger

### PR DESCRIPTION
**Pull request by pankajastro**
- Add DataprocBaseTrigger since other Dataproc triggers share lots of common params
- Move shared code to DataprocBaseTrigger
- Improve docstring

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by pankajastro** [comment]
Hi @potiuk can I get some reveiw on this please? Thank you!

**Review by potiuk — APPROVED** [review]
LGTM

File: airflow/providers/google/cloud/operators/dataproc.py [hunk 1]
```diff
@@ -876,12 +876,10 @@ def execute(self, context: Context) -> None:
                     project_id=self.project_id,
                     region=self.region,
                     cluster_name=self.cluster_name,
-                    request_id=self.request_id,
-                    retry=self.retry,
                     end_time=end_time,
                     metadata=self.metadata,
                     impersonation_chain=self.impersonation_chain,
-                    polling_interval=self.polling_interval_seconds,
+                    polling_interval_seconds=self.polling_interval_seconds,
```
Comment (potiuk) [inline airflow/providers/google/cloud/operators/dataproc.py:883]: Just one question - I guess this is in trigger so there are no worries about back-compatibility ?

Comment (pankajastro) [reply airflow/providers/google/cloud/operators/dataproc.py:883]: yes, I think that is correct.

**Pull request merged**